test/integration_test/basic_test.go

```diff
@@ -208,3 +221,70 @@ func BasicUpgrade(tc *TestCase) func(*testing.T) {
 		require.NoError(t, err)
 	}
 }
+
+func InstallWithTelemetry(tc *TestCase) func(*testing.T) {
+	return func(t *testing.T) {
+		if tc.ShouldSkip() {
+			t.Skip()
+		}
+
+		testSpec := tc.TestSpec(t)
+		cluster, ok := testSpec.(*op_corev1.StorageCluster)
+		require.True(t, ok)
+
+		testInstallWithTelemetry(t, cluster)
+	}
+}
+
+func testInstallWithTelemetry(t *testing.T, cluster *op_corev1.StorageCluster) {
+	secret := testutil.GetExpectedSecret(t, "/specs/pure-telemetry-cert.yaml")
+	require.NotNil(t, secret)
+
+	_, err := coreops.Instance().GetSecret(component.TelemetryCertName, "kube-system")
+	require.True(t, err == nil || errors.IsNotFound(err))
+
+	if errors.IsNotFound(err) {
+		secret, err = coreops.Instance().CreateSecret(secret)
+		require.NoError(t, err)
+	} else {
+		secret, err = coreops.Instance().UpdateSecret(secret)
+		require.NoError(t, err)
+	}
+
+	// Deploy portworx with telemetry set to true
+	imageListMap, err := testutil.GetImagesFromVersionURL(pxSpecGenURL)
+	require.NoError(t, err)
+
+	err = constructStorageCluster(cluster, pxSpecGenURL, imageListMap)
+	require.NoError(t, err)
```

Comment:
those are no longer needed, I initialize `pxSpecImages ` as a global variable during test setup, so we just need to call  `constructStorageCluster` in test spec